src/App.tsx

```diff
@@ -1,24 +1,214 @@
+  = useState(MakeTodosCompleted.begin);
+  const [clearCompleted, setClearCompleted] = useState(false);
+
+  useEffect(() => {
+    const filteredTodos = todos.filter((todo) => todo.completed);
+
+    setQuantity(todos.length - filteredTodos.length);
+  }, [todos]);
+
+  useEffect(() => {
+    setErrorMessage('');
+
+    todosService
+      .getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+      });
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    const filteredByTodos = todos.length
+      ? todos.filter((todo) => {
+        switch (filterBy) {
+          case Filter.active:
+            return !todo.completed;
+          case Filter.completed:
+            return todo.completed;
+          default:
+            return true;
+        }
+      })
+      : [];
+
+    return filteredByTodos;
+  }, [filterBy, todos]);
+
+  const deleteTodo = (todoId: number) => {
+    setErrorMessage('');
+    setIsFocused(false);
+
+    return todosService
+      .deleteTodo(todoId)
+      .then(() => {
+        setTodos((currentTodos) => currentTodos
+          .filter((todo) => todo.id !== todoId));
+      })
+      .catch((error) => {
+        setErrorMessage('Unable to delete a todo');
+        throw error;
+      })
+      .finally(() => {
+        setIsFocused(true);
+      });
+  };
+
+  const createNewTodo = ({ title, completed, userId }: Omit<Todo, 'id'>) => {
+    setErrorMessage('');
+
+    return todosService
+      .addTodo({ title, completed, userId })
+      .then((newTodo: Todo) => {
+        setTodos((currentTodos) => [...currentTodos, newTodo]);
+      })
+      .catch((error) => {
+        setErrorMessage('Unable to add a todo');
+        throw error;
+      });
+  };
+
+  const updateTodo = (updatedTodo: Todo) => {
+    setErrorMessage('');
+
+    return todosService
+      .updateTodo(updatedTodo)
+      .then((todo) => {
+        setTodos((currentTodos) => {
+          const newTodos = [...currentTodos];
+          const index = newTodos.findIndex(
+            (post) => post.id === updatedTodo.id,
+          );
+
+          newTodos.splice(index, 1, todo);
+
+          return newTodos;
+        });
+      })
+      .catch((error) => {
+        setErrorMessage('Unable to update a todo');
+        throw error;
+      });
+  };
+
+  const searchCompletedTodos = () => {
+    const hasCompletedTodos = filteredTodos.some((todo) => todo.completed);
+
+    setHasCompleted(hasCompletedTodos);
+  };
+
+  useEffect(() => {
+    const hasCompletedTodos = filteredTodos.some((todo) => todo.completed);
+
+    setHasCompleted(hasCompletedTodos);
+  }, [filteredTodos, hasCompleted]);
+
+  useEffect(() => {
+    setHideNotification(!errorMessage.length);
+  }, [errorMessage]);
+
+  useEffect(() => {
+    const hasAllCompleted = filteredTodos.every(todo => todo.completed);
+
+    setAllTodosCompleted(hasAllCompleted);
+  }, [filteredTodos, allTodosCompleted]);
```

Comment:
why you need this effect? please consider to remove it because it looks unnecessary

Reply:
I need this effect because we set the result of these effect into state which i use in button that toggles all todos.  We need to know if all todos are completed and we click on button, all todos should be uncompleted. If they are all uncompleted and we click in button, all todos should be completed. And if some of buttons aren't completed and we click on button, all buttons should be completed. To do this logic, i need to know about this